var/spack/repos/builtin/packages/py-filelock/package.py

```diff
@@ -0,0 +1,51 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyFilelock(PythonPackage):
+    """This package contains a single module, which implements a platform
+    independent file lock in Python, which provides a simple way of
+    inter-process communication"""
+
+    homepage = "https://github.com/benediktschmitt/py-filelock"
+    url = "https://github.com/benediktschmitt/py-filelock/archive/v3.0.4.tar.gz"
+
+    version('3.0.4',  '3cafce82375c3b635f2c872acaf3a00b')
+    version('3.0.3',  'e4bd69f15ebcc6d5a3d684cea3694840')
+    version('3.0.1',  'cbf41ad3d89c89e2b752bc85b501dff6')
+    version('3.0.0',  '29d199e8998ac324d0d7cab7aa814943')
+    version('2.0.13', 'cdd0c4f3e905fbab76d1202ce8e8b454')
+    version('2.0.12', 'fffda24b6cfd459ea5d2d5c335e949e2')
+    version('2.0.11', '9e8cbbe18494d12647050bb32a7e624d')
+    version('2.0.10', '1791e72bb19e503fdd0f365fb8ce2a4d')
+    version('2.0.9',  'b0269e7f77a090cc0d5fc9cf5fbe6ac2')
+    version('2.0.8',  '939ec6d4e2ecdc353a1f27fc452d8e8c')
+
+    depends_on('py-setuptools', type='build')
```

Comment:
this package does not actually use setuptools to install (you can check `setup.py` does not do a `import setuptools`, rather it uses `distutils`

Reply:
Right, but we do an inline patch to have it use `setuptools` -- doesn't that make it a requirement?

Reply:
ah you are right! sorry

Reply:
On second thought: Why do you do this in the first place?

Reply:
The patch? Because we ran into an error installing it with `distutils` and it turns out that `distutils` is deprecated. I was encountering:

```
1 error found in build log:
     9     ==> '/usr/workspace/wsa/bhatia4/trials/spack/opt/spack/linux-rhel6-x86_64/gcc-4.6.1/python-2.7.15-elzddn43epwkv2rl5ajgzgtphk2djops/bin/python2.7' '-s' 'set
           up.py' '--no-user-cfg' 'install' '--prefix=/usr/workspace/wsa/bhatia4/trials/spack/opt/spack/linux-rhel6-x86_64/gcc-4.6.1/py-filelock-3.0.4-62x47y3h4q3muow
           7x6t63egjjqfaivx4' '--single-version-externally-managed' '--root=/'
     10    usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     11       or: setup.py --help [cmd1 cmd2 ...]
     12       or: setup.py --help-commands
     13       or: setup.py cmd --help
     14
  >> 15    error: option --single-version-externally-managed not recognized
```

So I simply patched it to use `setuptools`.

Reply:
@adamjstewart could you take a look at this?

Reply:
Tell me more about this bug you ran into using distutils. A lot of Spack packages rely on distutils. I don't think any packages patch it to use setuptools, although pip does something like that.

Reply:
The exception above occurs if you were to remove the ```def patch(self)``` method as I specified it. I talked with @becker33 who informed me that `distutils` was more or less deprecated anyway, and that a simple patch would get me around the issue and use `setuptools` which is the more recently accepted standard.

Reply:
Yes, but if you remove both the patch _and_ the `depends_on('py-setuptools')`, that error should not occur.

Reply:
I imagine I would have to add `py-distutils` to the dependencies. I'll give it a test and make changes accordingly.

Reply:
@adamjstewart @healther -- I've corrected the package to use `distutils`. I didn't realize that's what it defaulted to, it seems to be functioning now.